### DISCUSSION TITLE
What's the Best Practice to make an updatable Provider?

### DISCUSSION THREAD

**Discussion by fellow-human-x** [rrousselGit/riverpod#3310] [Q&A]
I have a Screen with a ListView of a Study Materials List Tiles, when a List Tile is being clicked, a navigation will be done to a Details Screen presenting the material's details, and there's a capability to change the current presented material from the Details Screen.

Simply, it could be done by receiving the Study Material Object when navigating to the Details Screen, and calling setState when we change the current Study Material.

So, how to achieve that with Riverpod?

I tried the following:
- using StateNotifier and passing null as the initial value, but failed to pass the initial Study Material.
- using the Family modifier to pass the initial Study Material, but failed to change the state (the Study Material).

I just wanted to make a separation of concerns by moving the state modification outside the widget, but became confused when got in this situation.

Any guidance on that (the best practice to approach the problem)?

**Comment by RandalSchwartz** [comment]
Have you read the material at riverpod.dev and Andrea Bizzotto's excellent collection of Riverpod tutorials: https://codewithandrea.com/tags/riverpod/?  It's important to understand where you are at before we incrementally assist.

**Reply by fellow-human-x** [reply]
Yes, I did, and I followed along with Andrea in his two sections of SM in the Flutter Foundation Course.

But I'm a little confused with this situation, as I'm not so familiar maybe with riverpod, and used to work with GetX and the Apis shipped with the flutter sdk.
